### ISSUE TITLE
Question on persistence

### ISSUE THREAD

**Issue by mwildbolz** [seaside1/omatic#5]
Maybe more a question than an issue?!?

What do I have to do to persist the main values within openHAB - so, to have the
- Total Energy(Mea)
- Total Cost(Mea)
- Tot.Running Time

persisted over openHAB restarts?
I already tried to persist the linked items via mysql (and restore on startup) but it doesn't work...

Thanks in advance,
mw

**Comment by mwildbolz** [comment]
> Hi!
> 
> It's persisted on each complete cycle, i.e when it goes to finished. I'll add a timer to persist it like every minute, and also on shutdown/restart etc. You can create proxy items and persist those with mysql or similar, but items linked to channels will have its values overwritten. The persistence of the binding is done with binding properties that are loaded on startup.

So, if I understand you correctly, I would have to create 2 items for every value I'd like to persist
- one linked to the channel of the binding
- the second one as proxy item, where I additionally have to write a rule which updates the second item and adds for example the last energy needed value?

I don't understand, why this is neccessary. Shouldn't this be handled via the binding? So that I only have to persist (and restore on startup) the one item which is linked to the binding via the appropriate channel...

Or, I didn't get what you meant. Maybe you could post a short example of how to persist e.g. the TotalEnergy value?

**Comment by seaside1** [comment]
Yes you can create a proxy item to persist as a workaround. I will add something to control so it is persisted based on some timer rather than if the cycle is completed.
